From df49e0e3083b0707e2cca5a5956b49f14ded078e Mon Sep 17 00:00:00 2001
From: Guojing Yuan <guojing@google.com>
Date: Thu, 21 Mar 2024 22:12:07 +0000
Subject: [PATCH] [DO NOT MERGE][CDM] Fix a security issue that allow 3p apps
 to skip prompt by setSkipPrompt

Fix: 329230490

Test: CTS
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:71418ecfa539b99d9bb0053d1de5060040bdf02f)
Merged-In: I6e4dd33cbf98293d7efa0a40c0668d6c5242059a
Change-Id: I6e4dd33cbf98293d7efa0a40c0668d6c5242059a
---
 .../server/companion/CompanionDeviceManagerService.java     | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java b/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
index eccfe2ecb0b2..d2b34cc2304c 100644
--- a/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
+++ b/services/companion/java/com/android/server/companion/CompanionDeviceManagerService.java
@@ -429,10 +429,8 @@ public class CompanionDeviceManagerService extends SystemService implements Bind
             mCallingPackage = callingPackage;
             request.setCallingPackage(callingPackage);
 
-            if (mayAssociateWithoutPrompt(callingPackage, userId)) {
-                Slog.i(LOG_TAG, "setSkipPrompt(true)");
-                request.setSkipPrompt(true);
-            }
+            request.setSkipPrompt(mayAssociateWithoutPrompt(callingPackage, userId));
+
             callback.asBinder().linkToDeath(CompanionDeviceManagerService.this /* recipient */, 0);
 
             AndroidFuture<String> fetchProfileDescription =
-- 
2.45.0.rc1.225.g2a3ae87e7f-goog

